src/app/pages/questions/components/question/descriptive-input/descriptive-input.component.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  ViewChild
+} from '@angular/core'
+import { DomSanitizer } from '@angular/platform-browser'
+
+import { InfoItem } from '../../../../../shared/models/question'
+
+let uniqueID = 0
+
+@Component({
+  selector: 'descriptive-input',
+  templateUrl: 'descriptive-input.component.html'
+})
+export class DescriptiveInputComponent implements OnInit, OnChanges {
+  @ViewChild('content') content
+
+  @Output()
+  valueChange: EventEmitter<number> = new EventEmitter<number>()
+
+  @Input()
+  text: string
+  @Input()
+  currentlyShown: boolean
+
+  value: number = null
+  uniqueID: number = uniqueID++
+  name = `info-${this.uniqueID}`
+  items: InfoItem[] = Array()
+  showScrollButton: boolean
+  sanitizedHtml: any
+
+  constructor(private sanitizer: DomSanitizer) {}
+
+  ngOnInit() {
+    this.sanitizedHtml = this.sanitizer.bypassSecurityTrustHtml(this.text)
```

Comment:
is this necessary? Since we are just embedding media content? just asking as this could have serious security implications.
From the doc-
```
It is not required (and not recommended) to bypass security if the value is safe, e.g. a URL that does not start with a suspicious protocol, or an HTML snippet that does not contain dangerous code. The sanitizer leaves safe values intact.
```

Reply:
Yeah for embedding Youtube videos the iframe HTML tag is used and iframes can't be directly inserted into the component without using this bypass method. I'll try to find a way to make this safer.

Reply:
I've made the changes. I've added `DomPurify` which allows us to sanitize the HTML but allow `iframe` tags, before bypassing the security.